interfaces/builtin/app_launch.go

```diff
@@ -0,0 +1,58 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const appLaunchSummary = `allows snaps to identify and launch other snaps`
+
+const appLaunchBaseDeclarationSlots = `
+  app-launch:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
```

Comment:
The base declaration looks good. It is arguably superprivileged, *but* a) the interface gives precisely what it says it does and b) there are no side-effects like cross-snap data exfiltration and the like because the launch is constrained to the dbus api which doesn't provide passing arguments to the snap, so on the face of it, it is fine as is. Please have @pedronis signoff on this. Note, I would be comfortable with the following to make it superprivileged as well:
```
const appLaunchBaseDeclarationPlugs = `
  personal-files:
    allow-installation: false
    deny-auto-connection: true
`

const appLaunchBaseDeclarationSlots = `
  personal-files:
    allow-installation:
      slot-snap-type:
        - core
    deny-auto-connection: true
`
```

Reply:
I would prefer for this to start superprivileged until we have worked out things. We can be less restrictive at a future point.

Reply:
I'm not familiar with "superprivileged". As far as the PR is concerned that's the edit suggested above?